client/store/modules/repository/activities/actions.js

```diff
@@ -55,7 +57,19 @@ const calculateInsertPosition = ({ state }, { activity, anchor, action }) => {
   return calculatePosition(context);
 };
 
+const calculateCopyInsertPosition = ({ state }, { anchor, action, copyIndex }) => {
+  let items = getOutlineChildren(state.items, anchor.parentId);
+  if (action === ADD_INTO) {
+    items = getOutlineChildren(state.items, anchor.id);
+    return calculatePosition({ items, action });
+  }
+  const newPosition = findIndex(items, { id: anchor.id }) + copyIndex;
```

Comment:
Can we use the newly created (from the previous loop) anchor/node instead of adding a copy index 🤔 ?

Reply:
@underscope refactored the code to use previously created activity as anchor. I would still keep both `calculateInsertPosition` and `calculateCopyInsertPosition` since using only `calculateInsertPosition` would increase complexity of the action and decrease readability.